### PR TITLE
Change the zookeeper set configuration

### PR DESCRIPTION
**Pull request by chaithu14**

### CODE REVIEW

File: contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java [hunk 1]
```diff
@@ -338,15 +338,20 @@ public void setTopic(String topic)
   public void setZookeeper(String zookeeperString)
```
Comment (tweise) [inline contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java]: The specification of the ZK ensemble should also support the syntax used in the core ZK api:

host1:port1,host2:port2,...

Now there is ambiguity when adding a cluster name, which we can address by using a different separator:

cluster1::host1:port1;cluster2::host2:port2

Please add a unit test for the parsing.

Also, I don't see a need to delegate getters or setters to the consumer. We will take this up as separate effort along with app builder.

Comment (chaithu14) [inline contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java]: Updated the zookeeper setup as per the comments and also added the unit test for parsing.

File: contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java [hunk 2]
```diff
@@ -338,15 +338,21 @@ public void setTopic(String topic)
...
   {
     SetMultimap<String, String> theClusters = HashMultimap.create();
-    for (String zk : zookeeperString.split(",")) {
-      String[] parts = zk.split(":");
-      if (parts.length == 3) {
-        theClusters.put(parts[0], parts[1] + ":" + parts[2]);
-      } else if (parts.length == 2) {
-        theClusters.put(KafkaPartition.DEFAULT_CLUSTERID, parts[0] + ":" + parts[1]);
+    for (String zk : zookeeperString.split(";")) {
+      String[] parts = zk.split("::");
```
Comment (siyuanh) [inline contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java]: You can make it simple as below:
zk = zk.contains("::")?KafkaPartition.DEFAULT_CLUSTERID+"::":"" + zk;
then you can use the code in first if block to handle

File: contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java [hunk 3]
```diff
@@ -338,15 +338,21 @@ public void setTopic(String topic)
...
+        }
+      } else if (parts.length == 1) {
+        String[] zkParts = parts[0].split(":");
+        for(String hostName : zkParts[0].split(",")) {
+          theClusters.put(KafkaPartition.DEFAULT_CLUSTERID, hostName + ":" + zkParts[1]);
+        }
       } else
         throw new IllegalArgumentException("Wrong zookeeper string: " + zookeeperString + "\n"
-            + " Expected format should be cluster1:zookeeper1:port1,cluster2:zookeeper2:port2 or zookeeper1:port1,zookeeper:port2");
+            + " Expected format should be cluster1::zookeeper1,zookeeper2:port1;cluster2::zookeeper3:port2 or zookeeper1:port1,zookeeper:port2");
```
Comment (siyuanh) [inline contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java]: comment is not accurate, port is mandatory for each zookeeper

Comment (tweise) [inline contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java]: Correct. But here the intention is to not repeat the port number for every host. Similar to what you see in HBase config.

**Comment by siyuanh** [comment]
Couple issues I found after merge 2 abstract kafka input operators:
1. There are compile errors in malhar benchmark project which depends on old class
2. Some clean up work needs to be done in SimpleKafkaConsumer:
  (1) Line 336 should be above line 333 and should be written in short form:
      if (metadataRefreshInterval <= 0) { return;}
  (2) line 102 kpl should be called kpS and it's better to construct a concurrent hash set in this constructor copy out of the pass-in set parameter
  (3) line 369 comment should be updated as well, and the line from 363-371 can be shorten to 
       if(b.equals(oldB)) {
           continue;
       } 
       // otherwise put to deltaPositive
       deltaPositive.put(b, kp);

**Pull request closed without merging**